packages/contracts-bedrock/test/vendor/Initializable.t.sol

```diff
@@ -225,242 +235,273 @@ contract Initializer_Test is Bridge_Initializer {
+                continue;
+            }
+
+            // Check if this contract is in the contracts array.
+            assertTrue(
+                _hasMatchingContract(contractName), string.concat("Missing ", contractName, " from contracts array")
+            );
+
+            // If the contract is proxied, check that the proxy is in the contracts array.
+            // Skip predeployed contracts for now since we don't yet keep track of the
+            // implementations inside of the deploy script. We should add support for this in the
+            // future so that we can properly check that the implementations for predeployed
+            // contracts are initialized too.
+            if (ForgeArtifacts.isProxiedContract(contractName) && !ForgeArtifacts.isPredeployedContract(contractName)) {
+                assertTrue(
+                    _hasMatchingContract(string.concat(contractName, "Proxy")),
+                    string.concat("Missing ", contractName, "Proxy from contracts array")
+                );
             }
         }
 
-        // Then get L2 contracts
-        command[2] = string.concat(
-            Executables.find,
-            " src/L2 -type f -exec basename {} \\;",
-            " | ",
-            Executables.sed,
-            " 's/\\.[^.]*$//'",
-            " | ",
-            Executables.jq,
-            " -R -s 'split(\"\n\")[:-1]'"
-        );
-        string[] memory l2ContractNames = abi.decode(vm.parseJson(string(Process.run(command))), (string[]));
+        // Attempt to re-initialize all contracts within the `contracts` array.
+        for (uint256 i; i < contracts.length; i++) {
+            InitializeableContract memory _contract = contracts[i];
+            string memory name = _getRealContractName(_contract.name);
 
-        for (uint256 i; i < l2ContractNames.length; i++) {
-            string memory contractName = l2ContractNames[i];
-            string memory contractAbi = ForgeArtifacts.getAbi(contractName);
+            // Grab the value of the "initialized" storage slot. Must handle special case for the
+            // FaultDisputeGame and PermissionedDisputeGame contracts since these have a different
+            // name for the "initialized" storage slot and are currently not properly labeled in
+            // the deployment script.
+            // TODO: Update deployment script to properly label the dispute game contracts.
+            uint8 initializedSlotVal;
+            if (LibString.eq(name, "FaultDisputeGame") || LibString.eq(name, "PermissionedDisputeGame")) {
+                StorageSlot memory slot = ForgeArtifacts.getInitializedSlot(name);
+                bytes32 slotVal = vm.load(_contract.target, bytes32(vm.parseUint(slot.slot)));
+                initializedSlotVal = uint8((uint256(slotVal) >> (slot.offset * 8)) & 0xFF);
+            } else {
+                initializedSlotVal = deploy.loadInitializedSlot(name);
+            }
```

Comment:
Since `loadInitializedSlot` takes a name arg, it seems this conditional might be better suited to live directly within the `loadInitializedSlot` method, that this whole block simplifies to 
```suggestion
            initializedSlotVal = deploy.loadInitializedSlot(name);
```

Reply:
Will update that

Reply:
Honestly I tried doing this and it ended up being really ugly because you need to use `_contract.target` for the FDG and PDG but you use the loaded address from artifacts for everything else.

Reply:
Perhaps `loadInitializedSlot` for certain contract names then? Just want to make sure there's no footguns in that method since it's not universally accurate. Though actually it ends up calling `ForgeArtifacts.getInitializedSlot` which now does special case the games and handle them, so maybe this method already works for them?

Reply:
Ok for now I just made this revert for `FaultDisputeGame` and `PermissionedDisputeGame`.